src/SNLS_cuda_portability.h

```diff
@@ -45,7 +48,7 @@ extern void CUDART_Check(const cudaError_t err, const char *file, const char *fu
 // to filter code that cannot be compiled for the device.
 //----------------------------------------------------------------------------------------
 
-#ifdef __CUDA_ARCH__
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ > 0)) || defined(__HIP_DEVICE_COMPILE__)
 #define __cuda_device_only__      
 #else
 #define __cuda_host_only__      
```

Comment:
So, we should probably move to something more generic here like `__gpu_device_only__` and `__gpu_host_only__` and maybe add the `snls` name some where in there as well just to avoid name clashing with other people's macros.

Reply:
Along a similar set of lines, it would also be good to update the device forall portion of the code.